OrchardCore.AuditTrail/Handlers/UserEventHandler.cs

```diff
@@ -1,132 +1,146 @@
-            RecordAuditTrailEventAsync(UserAuditTrailEventProvider.Created, context.User);
-
-        public Task DeletedAsync(UserContext context) =>
-            RecordAuditTrailEventAsync(UserAuditTrailEventProvider.Deleted, context.User);
-
-        #region Unused events
-
-        public Task RecoveringPasswordAsync(Action<string, string> reportError) =>
-            Task.CompletedTask;
-
-        public Task ResettingPasswordAsync(Action<string, string> reportError) =>
-            Task.CompletedTask;
-
-        public Task LoggingInAsync(string userName, Action<string, string> reportError) =>
-            Task.CompletedTask;
-
-        public Task RegistrationValidationAsync(Action<string, string> reportError) =>
-            Task.CompletedTask;
-
-        #endregion
-
-
-        private async Task RecordAuditTrailEventAsync(string eventName, string username)
-        {
-            if (string.IsNullOrEmpty(username)) return;
-
-            var userManager = GetUserManagerFromHttpContext();
-            var user = await userManager.FindByNameAsync(username);
-
-            if (user == null)
-            {
-                var eventData = new Dictionary<string, object>
-                {
-                    { "UserName", username }
-                };
-
-                await _auditTrailManager.AddAuditTrailEventAsync<UserAuditTrailEventProvider>(
-                    new AuditTrailContext(eventName, username, eventData, "user", username));
-            }
-            else
-            {
-                var eventData = new Dictionary<string, object>
-                {
-                    { "UserId", await userManager.GetUserIdAsync(user) },
-                    { "UserName", user.UserName }
-                };
-
-                await _auditTrailManager.AddAuditTrailEventAsync<UserAuditTrailEventProvider>(
-                    new AuditTrailContext(eventName, user.UserName, eventData, "user", user.UserName));
-            }
-        }
-
-        private async Task RecordAuditTrailEventAsync(string eventName, IUser user)
-        {
-            var userName = user.UserName;
-            var userManager = GetUserManagerFromHttpContext();
-
-            var eventData = new Dictionary<string, object>
-            {
-                { "UserId", await userManager.GetUserIdAsync(user) },
-                { "UserName", userName }
-            };
-
-            await _auditTrailManager.AddAuditTrailEventAsync<UserAuditTrailEventProvider>(
-                new AuditTrailContext(eventName,
-                    _hca.GetCurrentUserName() ?? userName,
-                    eventData, "user", userName));
-        }
-
-        // Need to resolve the UserManager from the HttpContext to prevent circular dependency. 
-        private UserManager<IUser> GetUserManagerFromHttpContext() =>
-            _hca.HttpContext.RequestServices.GetRequiredService<UserManager<IUser>>();
-    }
-}
+//using System;
+//using System.Collections.Generic;
```

Comment:
Why commented out VS remove?

Reply:
Ideally this module shouldn't be used at the moment. I can only imagine it was added to the solution in order to be evaluted. This has been superceded by a much more integrated current version.